src/userInputParameters/load_BC_list.cc

```diff
@@ -1,102 +1,78 @@
+                newBC.var_BC_type.push_back(DIRICHLET);
+                std::string dirichlet_val = temp[j].substr(10,temp[j].size());
+                dirichlet_val = dealii::Utilities::trim(dirichlet_val);
+                newBC.var_BC_val.push_back(dealii::Utilities::string_to_double(dirichlet_val));
             }
-          else if (boost::iequals(temp[i].substr(0, 7), "NEUMANN"))
-            {
-              newBC.var_BC_type.push_back(NEUMANN);
-              std::string neumann_val = temp[i].substr(8, temp[i].size());
-              neumann_val             = dealii::Utilities::trim(neumann_val);
-              newBC.var_BC_val.push_back(
-                dealii::Utilities::string_to_double(neumann_val));
+            else if (boost::iequals(temp[j].substr(0,7),"NEUMANN")){
+                newBC.var_BC_type.push_back(NEUMANN);
+                std::string neumann_val = temp[j].substr(8,temp[j].size());
+                neumann_val = dealii::Utilities::trim(neumann_val);
+                newBC.var_BC_val.push_back(dealii::Utilities::string_to_double(neumann_val));
             }
-          else
-            {
-              std::cout << temp[i].substr(0, 8) << std::endl;
-              std::cout << "Error: Boundary conditions specified improperly."
-                        << std::endl;
-              abort();
+            else {
+                std::cout << temp[j].substr(0,8) << std::endl;
+                std::cout << "Error: Boundary conditions specified improperly." << std::endl;
+                abort();
             }
-        }
-      BC_list.push_back(newBC);
 
-      // Validate input using something like this:
-      // try{
-      //     if ((BC_type_dim1_min == "PERIODIC") && (BC_type_dim1_max !=
-      //     "PERIODIC")){
-      //         throw 0;
-      //     }
-      //     if ((BC_type_dim2_min == "PERIODIC") && (BC_type_dim2_max !=
-      //     "PERIODIC")){
-      //         throw 0;
-      //     }
-      //     if ((BC_type_dim3_min == "PERIODIC") && (BC_type_dim3_max !=
-      //     "PERIODIC")){
-      //         throw 0;
-      //     }
-      // }
-      // catch (int e){
-      //     if (e == 0){
-      //         std::cout << "Error: For periodic BCs, both faces for a given
-      //         direction must be set as periodic. "
-      //                 "Please check the BCs that are set in ICs_and_BCs.h."
-      //                 << std::endl;
-      //     }
-      //     abort();
-      // }
+            // If periodic BCs are used, ensure they are applied on both sides of domain
+            if (j%2 == 0){
+                if ( (boost::iequals(temp[j],"PERIODIC") && !boost::iequals(temp[j+1],"PERIODIC")) || 
+                     (!boost::iequals(temp[j],"PERIODIC") && boost::iequals(temp[j+1],"PERIODIC")) ){
+                    std::cout << "Error: Periodic boundary condition must be specified on both sides of domain." << std::endl;
+                    abort();
```

Comment:
Again, I think `AssertThrow`should be used instead of `cout`

Reply:
I originally used AssertThrow but didn't like that is wouldn't display in release mode. Are there ever cases when we want error messages to display in release mode? Or do you recommend we don't use release mode?

Reply:
I don't think there are any deal.II asserts that will display in release mode. However, I think this is fine as you should initially run your program in debug mode to make sure everything is working as intended (hence the name debug mode).
Then, you can use release mode.

Reply:
I think the only time we want release mode errors are when a solution becomes NaN or something like a divide by zero. Basically anything that would happen after a few timesteps.

Reply:
Well my thought with this particular bug is that a user can specify a natural BC at x=0 and a periodic BC at x=L and the simulation will run to completion and they won't have any idea something was wrong until they visualize the data. At that point it might as well be NaN...